### PR TITLE
Add online.athn.browser.gnome

### PR DESCRIPTION
**Pull request by itzgoldenleonard**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: online.athn.browser.gnome.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        "--share=ipc",
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--share=network"
+    ],
+    "build-options": {
+        "prepend-path": "/usr/lib/sdk/rust-stable/bin",
+        "build-args": [
+            "--share=network"
```
Comment (hfiguiere) [inline online.athn.browser.gnome.json:20]: build must be offline, this is not allowed, and will fail.

File: online.athn.browser.gnome.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            "buildsystem": "simple",
+            "build-commands": [
+                "cargo build --release",
+                "install -D target/release/reference-browser-gnome /app/bin/reference-browser-gnome",
+                "mkdir -p /app/share/glib-2.0/schemas /app/share/applications /app/share/metainfo /app/share/icons/hicolor/scalable/apps",
+                "cp src/resources/online.athn.browser.gnome.gschema.xml /app/share/glib-2.0/schemas/",
+                "glib-compile-schemas /app/share/glib-2.0/schemas/",
+                "cp metadata/online.athn.browser.gnome.desktop /app/share/applications/online.athn.browser.gnome.desktop",
+                "cp metadata/online.athn.browser.gnome.metainfo.xml /app/share/metainfo/online.athn.browser.gnome.metainfo.xml",
+                "cp metadata/online.athn.browser.gnome.svg /app/share/icons/hicolor/scalable/apps/online.athn.browser.gnome.svg"
```
Comment (hfiguiere) [inline online.athn.browser.gnome.json:39]: lines37-39: if you use the `install` command, then you don't need line 34. (also applied to line 35)

File: online.athn.browser.gnome.json (hunk 1 above)
Comment (hfiguiere) [reply online.athn.browser.gnome.json:20]: see for the tools that helps if you don't have a tarball with vendored crates.

https://github.com/flatpak/flatpak-builder-tools/tree/master/cargo

File: online.athn.browser.gnome.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
+{
+    "app-id": "online.athn.browser.gnome",
```
Comment (hfiguiere) [inline online.athn.browser.gnome.json:2]: `browser.athn.online` doesn't exist.

File: online.athn.browser.gnome.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    ],
+    "build-options": {
+        "prepend-path": "/usr/lib/sdk/rust-stable/bin",
+        "build-args": [
+            "--share=network"
+        ],
+        "env": {
+            "RUST_BACKTRACE": "1"
+        },
+        "strip": true
```
Comment (hfiguiere) [inline online.athn.browser.gnome.json:25]: please don't. Debug packages with symbols are generated.

**Comment by hfiguiere** [comment]
bot, build online.athn.browser.gnome

File: online.athn.browser.gnome.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                "cp metadata/online.athn.browser.gnome.svg /app/share/icons/hicolor/scalable/apps/online.athn.browser.gnome.svg"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/itzgoldenleonard/reference-browser-gnome.git",
+                    "tag": "v0.1.0",
+                    "commit": "d8bc8efeec6c7430403b4dee72e29b7e1a09c250"
+                },
+                "metadata/cargo-sources.json"
```
Comment (hfiguiere) [inline online.athn.browser.gnome.json:46]: this file is missing.

**Comment by itzgoldenleonard** [comment]
bot, build online.athn.browser.gnome

**Comment by itzgoldenleonard** [comment]
bot, build online.athn.browser.gnome

File: online.athn.browser.gnome.json (hunk 2 above)
Comment (hfiguiere) [reply online.athn.browser.gnome.json:37]: you still should use install.

File: online.athn.browser.gnome.json [hunk 6]
```diff
@@ -29,12 +29,11 @@
                 "cargo --offline fetch --manifest-path Cargo.toml --verbose",
                 "cargo --offline build --release --verbose",
                 "install -Dm755 target/release/reference-browser-gnome -t /app/bin/reference-browser-gnome",
```
Comment (hfiguiere) [inline online.athn.browser.gnome.json:31]: this installs `/app/bin/reference-browser-gnome/reference-browser-gnome`

I doub't that's correct.

```suggestion
                "install -Dm755 target/release/reference-browser-gnome -t /app/bin",
```

**Pull request closed without merging**